packages/contract-wrappers/src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -0,0 +1,994 @@
+            },
+        );
+        return txHash;
+    }
+    /**
+     * Batch version of cancelOrderAsync. Executes multiple cancels atomically in a single transaction.
+     * @param   orders                                          An array of orders to cancel.
+     * @param   orderTransactionOpts                            Optional arguments this method accepts.
+     * @return  Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async batchCancelOrdersAsync(
+        orders: Order[],
+        orderTransactionOpts: OrderTransactionOpts = {},
+    ): Promise<string> {
+        assert.doesConformToSchema('orders', orders, schemas.ordersSchema);
+        assert.doesConformToSchema('orderTransactionOpts', orderTransactionOpts, orderTxOptsSchema, [txOptsSchema]);
+        const makerAddresses = _.map(orders, order => order.makerAddress);
+        const makerAddress = makerAddresses[0];
+        await assert.isSenderAddressAsync('makerAddress', makerAddress, this._web3Wrapper);
+        const normalizedMakerAddress = makerAddress.toLowerCase();
+
+        const exchangeInstance = await this._getExchangeContractAsync();
+        const txHash = await exchangeInstance.batchCancelOrders.sendTransactionAsync(orders, {
+            from: normalizedMakerAddress,
+            gas: orderTransactionOpts.gasLimit,
+            gasPrice: orderTransactionOpts.gasPrice,
+        });
+        return txHash;
+    }
+    /**
+     * Match two complementary orders that have a profitable spread.
+     * Each order is filled at their respective price point. However, the calculations are carried out as though
+     * the orders are both being filled at the right order's price point.
+     * The profit made by the left order goes to the taker (who matched the two orders).
+     * @param leftSignedOrder         First order to match.
+     * @param rightSignedOrder        Second order to match.
+     * @param takerAddress      The address that sends the transaction and gets the spread.
+     * @return Transaction hash.
+     */
+    @decorators.asyncZeroExErrorHandler
+    public async matchOrdersAsync(
+        leftSignedOrder: SignedOrder,
+        rightSignedOrder: SignedOrder,
+        takerAddress: string,
+        orderTransactionOpts: OrderTransactionOpts = {},
+    ): Promise<string> {
+        assert.doesConformToSchema('leftSignedOrder', leftSignedOrder, schemas.signedOrderSchema);
+        assert.doesConformToSchema('rightSignedOrder', rightSignedOrder, schemas.signedOrderSchema);
+        await assert.isSenderAddressAsync('takerAddress', takerAddress, this._web3Wrapper);
+        assert.doesConformToSchema('orderTransactionOpts', orderTransactionOpts, orderTxOptsSchema, [txOptsSchema]);
+        const normalizedTakerAddress = takerAddress.toLowerCase();
+        // TODO(logvinov): Check that:
+        // rightOrder.makerAssetData === leftOrder.takerAssetData;
+        // rightOrder.takerAssetData === leftOrder.makerAssetData;
```

Comment:
When will this TODO be addressed?

Reply:
In the new PR. Part 3. I'll add missing validations there. This one doesn't have any validations and I'm not sure if it makes sense to add just one.

Reply:
There will be also Part 4 that will assign those variables to NULL_ADDRESS to save gas.